### PR TITLE
New authentication method with Bearer.

### PR DESCRIPTION
**Pull request by Gnafu**
Related to #1926

### CODE REVIEW

File: web/client/actions/security.js [hunk 1]
```diff
@@ -18,13 +22,14 @@ const CHANGE_PASSWORD = 'CHANGE_PASSWORD';
...
 
 function loginSuccess(userDetails, username, password, authProvider) {
     return {
         type: LOGIN_SUCCESS,
         userDetails: userDetails,
         // set here for compatibility reasons
+        // TODO: verify if the compatibility reasons still hold and remove otherwise
         authHeader: 'Basic ' + btoa(username + ':' + password),
         username: username,
         password: password,
```
Comment (mbarto) [inline web/client/actions/security.js]: Is there a reason why we store the password on the state?

File: web/client/api/GeoStoreDAO.js [hunk 2]
```diff
@@ -65,6 +66,28 @@ var Api = {
             return response.data;
         });
     },
+    login: function(username, password, options) {
+        let url = "session/login";
```
Comment (mbarto) [inline web/client/api/GeoStoreDAO.js]: const

File: web/client/api/GeoStoreDAO.js [hunk 3]
```diff
@@ -65,6 +66,28 @@ var Api = {
...
+                headers: {
+                    'Authorization': 'Bearer ' + response.data.access_token
+                },
+                params: {
+                    includeattributes: true
+                }
+            }, options)));
+        }).then((response) => {
+            return { ...response.data, ...authData};
+        });
```
Comment (mbarto) [inline web/client/api/GeoStoreDAO.js]: Let's add some catch to handle errors

File: web/client/reducers/security.js [hunk 4]
```diff
@@ -68,6 +82,14 @@ function security(state = {user: null, errorCause: null}, action) {
                 passwordError: action.error,
                 passwordChanged: false
             });
+        case REFRESH_ACCESS_TOKEN:
```
Comment (mbarto) [inline web/client/reducers/security.js]: Not needed

File: web/client/actions/security.js (hunk 1 above)
Comment (Gnafu) [inline web/client/actions/security.js]: there are other places (in reducers) where the authHeader is computed. Don't know why, hence the question on "compatibility reasons"

Comment (mbarto) [inline web/client/actions/security.js]: ok

File: web/client/api/GeoStoreDAO.js (hunk 3 above)
Comment (Gnafu) [inline web/client/api/GeoStoreDAO.js]: Errors are catched in the action, see https://github.com/geosolutions-it/MapStore2/pull/1949/files#diff-cb95a3cd779691bf0b73f9f8eabed903L66

Comment (mbarto) [inline web/client/api/GeoStoreDAO.js]: ok

File: web/client/reducers/security.js [hunk 5]
```diff
@@ -6,7 +6,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-const { LOGIN_SUCCESS, LOGIN_FAIL, LOGOUT, CHANGE_PASSWORD_SUCCESS, CHANGE_PASSWORD_FAIL, RESET_ERROR } = require('../actions/security');
+const { LOGIN_SUCCESS, LOGIN_FAIL, LOGOUT, CHANGE_PASSWORD_SUCCESS, CHANGE_PASSWORD_FAIL, RESET_ERROR, REFRESH_ACCESS_TOKEN, REFRESH_SUCCESS } = require('../actions/security');
```
Comment (mbarto) [inline web/client/reducers/security.js]: Remove REFRESH_ACCESS_TOKEN to make ESLint happy

File: web/client/product/app.jsx [hunk 6]
```diff
@@ -32,7 +33,8 @@ const startApp = () => {
 
     const initialActions = [
         () => loadMaps(ConfigUtils.getDefaults().geoStoreUrl, ConfigUtils.getDefaults().initialMapFilter || "*"),
-        loadVersion
+        loadVersion,
+        verifySession
```
Comment (offtherailz) [inline web/client/product/app.jsx]: This is ok, but I'd like to have it connected directly to the plugin, as an epic. So I don't have to bring it in all installations.
Is it possible?

**Pull request merged**